compiler/rustc_hir_analysis/src/check/mod.rs

```diff
@@ -113,7 +113,17 @@ pub fn provide(providers: &mut Providers) {
 }
 
 fn adt_destructor(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<ty::Destructor> {
-    tcx.calculate_dtor(def_id, always_applicable::check_drop_impl)
+    let dtor = tcx.calculate_dtor(def_id, always_applicable::check_drop_impl);
+    if dtor.is_none() && tcx.features().async_drop() {
+        if let Some(async_dtor) = adt_async_destructor(tcx, def_id) {
+            // When type has AsyncDrop impl, but doesn't have Drop impl, generate error
+            let span = tcx.def_span(def_id.to_def_id());
+            let impl_span = tcx.def_span(async_dtor.impl_did);
```

Comment:
Make this the main span. Unsure if pointing to the type is useful, we could just leave it off entirely